parser-typechecker/src/Unison/Runtime/Interface.hs

```diff
@@ -696,23 +704,43 @@ nativeEvalInContext _ ctx codes base = do
       -- decodeResult . deserializeValue =<< BS.hGetContents pout
       callout _ _ _ _ =
         pure . Left $ "withCreateProcess didn't provide handles"
-  withCreateProcess (schemeProc []) callout
+      ucrError (_ :: IOException) =
+        die
+          "I had trouble calling the unison runtime exectuable.\n\n\
+          \Please check that the `ucr` executable is properly\
+          \ installed."
+  withCreateProcess (ucrProc []) callout `UnliftIO.catch` ucrError
 
 nativeCompileCodes ::
   [(Reference, SuperGroup Symbol)] ->
   Reference ->
   FilePath ->
   IO ()
 nativeCompileCodes codes base path = do
+  genDir <- getXdgDirectory XdgCache "unisonlanguage/racket-tmp"
+  createDirectoryIfMissing True genDir
   let bytes = serializeValue . compileValue base $ codes
+      srcPath = genDir </> path <.> "rkt"
       callout (Just pin) _ _ ph = do
         BS.hPut pin . runPutS . putWord32be . fromIntegral $ BS.length bytes
         BS.hPut pin bytes
         UnliftIO.hClose pin
         waitForProcess ph
         pure ()
       callout _ _ _ _ = fail "withCreateProcess didn't provide handles"
-  withCreateProcess (schemeProc ["-o", path]) callout
+      ucrError (_ :: IOException) =
+        die
+          "I had trouble calling the unison runtime exectuable.\n\n\
+          \Please check that the `ucr` executable is properly\
```

Comment:
Would be great to link to a guide in these errors, as an end user I wouldn't know where to start on ensuring "ucr" is installed and google probably won't help.

Reply:
Yeah, I'd like to make them better. I'm not exactly sure what the install process will be right now, though. Hopefully I can flesh these out after I work out some details with Arya.

Reply:
I'm anticipating that either our installers will set it up correctly for them, or they're in big trouble and need to come ask for help, which will help us flesh out better error messages :-\